src/components/Image/Image.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react';
+
+import { Skeleton } from '../Skeleton';
+
+import styles from './Image.module.scss';
+import classNames from 'classnames';
+
+export interface ImageProps {
+  alt: string;
+  src?: string;
+  className?: string;
+  objectFit?: 'contain' | 'cover' | 'fill' | 'none' | 'scale-down';
+  onLoad?: () => void;
+  onError?: () => void;
+}
+
+interface ImageState {
+  isCached: boolean;
+  src: string;
+}
+
+// named Img here to avoid conflict with the HTMLImageElement in useEffect
+const Img = ({ className, objectFit = 'cover', alt, src, onLoad, onError }: ImageProps) => {
+  const [image, setImage] = React.useState<ImageState | undefined>();
```

Comment:
nit: no need to specify `undefined` in state type.

Reply:
I didn't know that - thanks!